test/check-cli

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/python3
+
+import argparse
+import os
+import signal
+import subprocess
+import sys
+
+# import Cockpit's machinery for test VMs and its browser test API
+sys.path.append(os.path.join(os.path.dirname(__file__), "../bots/machine"))
+import testvm
+
+
+def main(image):
+    network = testvm.VirtNetwork(0)
+    machine = testvm.VirtMachine(image, networking=network.host(), memory_mb=2048)
+    machine.start()
+    machine.wait_boot()
+
+    # run a command to force starting the SSH master
+    machine.execute('uptime')
+
+    ssh_command = ["ssh", "-o", "ControlPath=" + machine.ssh_master,
+                          "-p", machine.ssh_port,
+                          machine.ssh_user + "@" + machine.ssh_address]
+
+    print(f"Virtual machine running {image} is up. Connect to it via:")
+    print(" ".join(ssh_command))
+    print()
+
+    r = subprocess.run(ssh_command + ["CLI=/usr/bin/composer-cli", "/tests/test_cli.sh"])
```

Comment:
or maybe `make test_images test_aws test_azure`, etc.

All of these require credentials for the various cloud providers. Is it possible to securely store these credentials in the new CI infrastructure? 

More importantly quickly updating the credentials (also via ENV/manually) when they change or when we want to execute tests with a different account ?

Reply:
Let's call the test scripts directly from here, as this script is already called from the Makefile. Too many circles :)

Yes, we can store credentials in the CI infrastructure. We already do this for GitHub tokens and the RHEL subscription. Are you ok to get the basic machinery working in this PR with only the tests that run without extra credentials. Then do a follow up with one that adds support for the cloud tests?